mailbox/store/src/main/java/org/apache/james/mailbox/store/mail/utils/MimeMessageHeadersUtil.java

```diff
@@ -0,0 +1,58 @@
+/******************************************************************
+ * Licensed to the Apache Software Foundation (ASF) under one     *
+ * or more contributor license agreements.  See the NOTICE file   *
+ * distributed with this work for additional information          *
+ * regarding copyright ownership.  The ASF licenses this file     *
+ * to you under the Apache License, Version 2.0 (the              *
+ * "License"); you may not use this file except in compliance     *
+ * with the License.  You may obtain a copy of the License at     *
+ *                                                                *
+ * http://www.apache.org/licenses/LICENSE-2.0                     *
+ *                                                                *
+ * Unless required by applicable law or agreed to in writing,     *
+ * software distributed under the License is distributed on an    *
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY         *
+ * KIND, either express or implied.  See the License for the      *
+ * specific language governing permissions and limitations        *
+ * under the License.                                             *
+ ******************************************************************/
+
+package org.apache.james.mailbox.store.mail.utils;
+
+import java.util.List;
+import java.util.Optional;
+import java.util.stream.Collectors;
+
+import org.apache.james.mailbox.store.mail.model.MimeMessageId;
+import org.apache.james.mailbox.store.mail.model.Subject;
+import org.apache.james.mime4j.message.HeaderImpl;
+import org.apache.james.mime4j.stream.Field;
+
+public class MimeMessageHeadersUtil {
+    public static MimeMessageId parseMimeMessageId(HeaderImpl headers) {
+        return new MimeMessageId(headers.getField("Message-ID").getBody());
+    }
+
+    public static Optional<MimeMessageId> parseInReplyTo(HeaderImpl headers) {
+        Field inReplyToField = headers.getField("In-Reply-To");
+        if (inReplyToField != null) {
+            return Optional.of(new MimeMessageId(inReplyToField.getBody()));
+        }
+        return Optional.empty();
+    }
+
+    public static Optional<List<MimeMessageId>> parseReferences(HeaderImpl headers) {
+        List<Field> mimeMessageIdFields = headers.getFields("References");
+        if (!mimeMessageIdFields.isEmpty()) {
+            List<MimeMessageId> mimeMessageIdList = mimeMessageIdFields.stream()
+                .map(mimeMessageIdField -> new MimeMessageId(mimeMessageIdField.getBody()))
+                .collect(Collectors.toList());
+            return Optional.of(mimeMessageIdList);
+        }
+        return Optional.empty();
+    }
+
+    public static Subject parseSubject(HeaderImpl headers) {
+        return new Subject(headers.getField("Subject").getBody());
```

Comment:
`UnstructuredFieldImpl::getValue`

(We need to handle possible encoded characters IMO cf HeaderCollectionTest::getHeadersShouldDecodeValues)